packages/misc/commands/dm.js

```diff
@@ -47,37 +47,64 @@ module.exports = {
 			isSilent = interaction.options.getBoolean("silent");
 		}
 		*/
-		const msg = interaction.options.getString(
-			LANG.commands.dm.options.text.name,
-		);
 
 		const userId = interaction.options.getUser(
 			LANG.commands.dm.options.user.name,
 		);
-		const dmChannel = await userId.createDM();
+		if (userId === interaction.user.bot) {
+			await interaction.reply({
+				content: 'Botに対して実行することはできません!',
+				ephemeral: true,
+			});
+		}
 
-		dmChannel.send({
-			embeds: [
-				{
-					title: strFormat(LANG.commands.dm.messageTitle, [
-						interaction.user.username,
-					]),
-					thumbnail: {
-						url: interaction.user.displayAvatarURL(),
-					},
-					color: 0x5865f2,
-					fields: [
-						{
-							name: LANG.commands.dm.messageFieldName,
-							value: msg,
-						},
-					],
-				},
-			],
-		});
+		const modal = new ModalBuilder()
+			.setCustomId('modaldm')
+			.setTitle('DMを送信します...');
+
+		const msgcontent = new TextInputBuilder()
+			.setCustomId('content')
+			.setLabel('送信したいメッセージ')
+			.setStyle(TextInputStyle.Paragraph)
+			.setPlaceholder('送りたいメッセージをここに記入...')
+			.setRequired(true)
+			.setMaxLength(1024);
+		const firstRow = new ActionRowBuilder().addComponents(msgcontent);
+		modal.addComponents(firstRow);
+		await interaction.showModal(modal);
+		// const filter = (mInteraction) => mInteraction.customId === 'gbanreport';
+		const submitted = await interaction
+			.awaitModalSubmit({
+				time: 60000,
+				filter: (i) => i.user.id === interaction.user.id,
+			})
+			.catch((e) => console.error(e));
```

Comment:
`try` `catch` `finally` 使いましょう！
```suggestion
        try {
            await interaction.awaitModalSubmit({
                time: 60000,
                filter: (i) => i.user.id === interaction.user.id,
            });
            // if (submitted) { ... } 内のコード
        } catch (e) {
            (e) => console.error(e)
        } finally {
            // ↓ 成功した場合、失敗した場合で共通のコード
            const expirationTime = Date.now() + cooldownTime * 1000;
            cooldowns.set(executorId, expirationTime);
        }
```

Reply:
あっこれdiff形式になってるけど正確じゃないから許して